### PR TITLE
Mips new opcodes + bugfix

### PR DESCRIPTION
**Pull request by IridiumXOR**
Hi,
I have implemented this opcodes for MIPS
TNE, CLZ, LL, SC, SYNC, PREF, TLBR, MTHI, MTLO

I have also introduced this opcodes (only disasm)
CACHE, ERET

I have fix a typo (KSCRATCH0 -> KSCRATCH) and a little bug on opcode TEQ

### CODE REVIEW

File: miasm/arch/mips32/sem.py [hunk 1]
```diff
@@ -488,6 +509,66 @@ def ei(arg1):
...
+@sbuild.parse
+def mthi(arg1):
+    R_HI = arg1
+
+@sbuild.parse
+def mtlo(arg1):
+    R_LOW = arg1
+
+@sbuild.parse
+def clz(rs, rd):
```
Comment (serpilliere) [inline miasm/arch/mips32/sem.py:526]: Maybe you can use the miasm operator "cntlz". For example, in arm, we have:
```python
def clz(ir, instr, a, b):
    e = []
    e.append(ExprAssign(a, ExprOp('cntleadzeros', b)))
    return e, []
```

Comment (IridiumXOR) [reply miasm/arch/mips32/sem.py]: I have replace it with 'cntleadzeros'

Comment (IridiumXOR) [reply miasm/arch/mips32/sem.py]: However, a "manual" implementation of clz (from which I have copied my past implementation) it is present in arch/mep/sem.py

Comment (serpilliere) [reply miasm/arch/mips32/sem.py]: Oh! I think I did a bad review of the mep before merging it.

**Comment by serpilliere** [comment]
@IridiumXOR : the PR seems nearly ok to me:
Can you:
- merge your 2 commits together (to avoid having the fixing commit in the master)
- add another commit with some asm/disasm regression tests?

Tell me if you don't have time to do this: I will merge your PR and add tests later.

**Comment by IridiumXOR** [comment]
Sorry, I was not able to merge commits :( I'm a noob with git..

**Comment by serpilliere** [comment]
Noproblemo
Thank you for the fix @IridiumXOR !

**Pull request merged**